google-cloud-contrib/google-cloud-nio/src/main/java/com/google/cloud/storage/contrib/nio/SeekableByteChannelPrefetcher.java

```diff
@@ -0,0 +1,474 @@
+   * the entity to grow to accommodate the new bytes; the values of any bytes
+   * between the previous end-of-file and the newly-written bytes are
+   * unspecified.
+   *
+   * <p> Setting the channel's position is not recommended when connected to
+   * an entity, typically a file, that is opened with the
+   * StandardOpenOption.APPEND option. When opened for
+   * append, the position is first advanced to the end before writing.
+   *
+   * @param newPosition The new position, a non-negative integer counting
+   *                    the number of bytes from the beginning of the entity
+   * @return This channel
+   * @throws ClosedChannelException   If this channel is closed
+   * @throws IllegalArgumentException If the new position is negative
+   * @throws IOException              If some other I/O error occurs
+   */
+  @Override
+  public SeekableByteChannel position(long newPosition) throws IOException {
+    if (!open) {
+      throw new ClosedChannelException();
+    }
+    position = newPosition;
+    return this;
+  }
+
+  /**
+   * Returns the current size of entity to which this channel is connected.
+   *
+   * @return The current size, measured in bytes
+   * @throws ClosedChannelException If this channel is closed
+   * @throws IOException            If some other I/O error occurs
+   */
+  @Override
+  public long size() throws IOException {
+    if (!open) {
+      throw new ClosedChannelException();
+    }
+    return size;
+  }
+
+  /**
+   * Not supported.
+   */
+  @Override
+  public SeekableByteChannel truncate(long size) throws IOException {
+    throw new NonWritableChannelException();
+  }
+
+  /**
+   * Tells whether or not this channel is open.
+   *
+   * @return <tt>true</tt> if, and only if, this channel is open
+   */
+  @Override
+  public boolean isOpen() {
+    return open;
+  }
+
+  /**
+   * Closes this channel.
+   *
+   * <p> After a channel is closed, any further attempt to invoke I/O
+   * operations upon it will cause a {@link ClosedChannelException} to be
+   * thrown.
+   *
+   * <p> If this channel is already closed then invoking this method has no
+   * effect.
+   *
+   * <p> This method may be invoked at any time.  If some other thread has
+   * already invoked it, however, then another invocation will block until
+   * the first invocation is complete, after which it will return without
+   * effect. </p>
+   *
+   * @throws IOException If an I/O error occurs
+   */
+  @Override
+  public void close() throws IOException {
+    if (open) {
+      closing = true;
+      exec.shutdown();
+      try {
+        while (true) {
+            synchronized (idleWorkers) {
+            if (idleWorkers.size() >= workerCount) {
+              // every thread is idle, we're done.
+              break;
+            }
+            idleWorkers.wait(60_000);
+          }
+        }
+      } catch (InterruptedException e) {
+        System.out.println("Timed out while waiting for channels to close.");
```

Comment:
You need to either rethrow the InterruptedException or set the interrupted flag to true.